nodebuilder/nodebuilder.go

```diff
@@ -110,60 +107,63 @@ func (nb *NodeBuilder) StartTracing() {
 }
 
 func (nb *NodeBuilder) startSigner(ctx context.Context) error {
-	localKeys := nb.Config.PrivateKeySet
-	localSigner := types.NewLocalSigner(&localKeys.DestKey.PublicKey, localKeys.SignKey)
-
-	currentPath := signerCurrentPath(nb.Config.StoragePath, localSigner)
-
-	middleware.Log.Debugw("starting signer node", "storagePath", currentPath)
-	badgerCurrent, err := storage.NewDefaultBadger(currentPath)
-	if err != nil {
-		return fmt.Errorf("error creating storage: %v", err)
-	}
-
-	group, err := nb.Config.NotaryGroupConfig.NotaryGroup(localSigner)
-	if err != nil {
-		return fmt.Errorf("error generating notary group: %v", err)
-	}
-
-	var pubsub remote.PubSub
-	remote.Start()
-
-	cm := connmgr.NewConnManager(len(group.Signers)*2, 900, 20*time.Second)
-	for _, s := range group.Signers {
-		id, err := p2p.PeerFromEcdsaKey(s.DstKey)
-		if err != nil {
-			panic(fmt.Sprintf("error getting peer from ecdsa key: %v", err))
-		}
-		cm.Protect(id, "signer")
-	}
-
-	p2pHost, err := nb.p2pNodeWithOpts(ctx, p2p.WithLibp2pOptions(libp2p.ConnectionManager(cm)))
-	if err != nil {
-		return fmt.Errorf("error setting up p2p host: %v", err)
-	}
-	if _, err = p2pHost.Bootstrap(nb.bootstrapNodes()); err != nil {
-		return fmt.Errorf("failed to bootstrap: %s", err)
-	}
-
-	remote.NewRouter(p2pHost)
-
-	nb.host = p2pHost
-
-	pubsub = remote.NewNetworkPubSub(p2pHost.GetPubSub())
-
-	act, err := actor.EmptyRootContext.SpawnNamed(actors.NewTupeloNodeProps(&actors.TupeloConfig{
-		Self:              localSigner,
-		NotaryGroup:       group,
-		CurrentStateStore: badgerCurrent,
-		PubSubSystem:      pubsub,
-	}), syncerActorName(localSigner))
-	if err != nil {
-		panic(fmt.Sprintf("error spawning: %v", err))
-	}
-
-	localSigner.Actor = act
-	nb.actorToStop = act
+	// TODO: Commented out pending gossip4. We may or may not want to convert this.
```

Comment:
are we doing something else to start the gossip4 signers?

Reply:
Yes, `cmd/node.go` just directly initializes gossip4 nodes (via the code in `gossip4/node.go`) now. It might make sense to reintroduce something like `nodebuilder` for gossip4 though.